pkg/git/gogit.go

```diff
@@ -34,15 +35,28 @@ import (
 	"github.com/weaveworks/weave-gitops/pkg/utils"
 )
 
+type GotGit struct{}
+
+func (g *GotGit) PlainCloneContext(ctx context.Context, path string, isBare bool, o *gogit.CloneOptions) (*gogit.Repository, error) {
+	return gogit.PlainCloneContext(ctx, path, isBare, o)
+}
+
+//counterfeiter:generate . GoGitI
+type GoGitI interface {
+	PlainCloneContext(ctx context.Context, path string, isBare bool, o *gogit.CloneOptions) (*gogit.Repository, error)
+}
+
 type GoGit struct {
 	path       string
 	auth       transport.AuthMethod
 	repository *gogit.Repository
+	gitwrapper GoGitI
 }
 
-func New(auth transport.AuthMethod) *GoGit {
+func New(auth transport.AuthMethod, wrapper GoGitI) *GoGit {
```

Comment:
This seems strange. We are returning a `*GoGit` but also passing in a `*GoGit`. What is the goal here?